NullPointerException on JavaDocParserVisitor for classes without javadoc

When a resource doesn't have javadoc comment at class level but has **any field** with javadoc then   JavadocParser [visit method](https://github.com/sdaschner/jaxrs-analyzer/blob/master/src/main/java/com/sebastian_daschner/jaxrs_analyzer/analysis/javadoc/JavaDocParserVisitor.java#L90) will raise a Nullpointer because `classComments.get(className)` returns null. Following is an example:

```
@Stateless
@Path("resourcesPath")
public class JaxRSResources {

  /**
   * a simple javadoc comment
   */
  @Inject
  ApplicationProperties applicationProperties;  
```
The stacktrace is below:
```
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal com.sebastian-daschner:jaxrs-analyzer-maven-plugin:0.17-SNAPSHOT:analyze-jaxrs failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at com.sebastian_daschner.jaxrs_analyzer.analysis.javadoc.JavaDocParserVisitor.lambda$visit$1 (JavaDocParserVisitor.java:90)
    at java.util.Optional.ifPresent (Optional.java:159)
    at com.sebastian_daschner.jaxrs_analyzer.analysis.javadoc.JavaDocParserVisitor.visit (JavaDocParserVisitor.java:90)
    at com.sebastian_daschner.jaxrs_analyzer.analysis.javadoc.JavaDocParserVisitor.visit (JavaDocParserVisitor.java:36)

```

Analyzer version: 0.17-SNAPSHOT

--- a/src/main/java/com/sebastian_daschner/jaxrs_analyzer/analysis/javadoc/JavaDocParserVisitor.java
+++ b/src/main/java/com/sebastian_daschner/jaxrs_analyzer/analysis/javadoc/JavaDocParserVisitor.java
@@ -87,12 +87,17 @@ public class JavaDocParserVisitor extends VoidVisitorAdapter<Void> {
         field.getComment()
                 .filter(Comment::isJavadocComment)
                 .map(this::toJavaDoc)
-                .ifPresent(c -> classComments.get(className).getFieldComments().add(createFieldComment(c, field)));
+                .ifPresent(c ->createFieldComment(c, field));
         super.visit(field, arg);
     }
 
-    private MemberParameterTag createFieldComment(Javadoc javadoc, FieldDeclaration field) {
-        return createMemberParamTag(javadoc.getDescription(), field.getAnnotations().stream());
+    private void createFieldComment(Javadoc javadoc, FieldDeclaration field) {
+      ClassComment classComment = classComments.get(className);
+      if(classComment == null) {
+        classComment = new ClassComment("", new HashMap<>(), false);
+        classComments.put(className, classComment);
+      }
+      classComment.getFieldComments().add(createMemberParamTag(javadoc.getDescription(), field.getAnnotations().stream()));
     }
 
     @Override

--- a/src/test/java/com/sebastian_daschner/jaxrs_analyzer/analysis/ProjectAnalyzerTest.java
+++ b/src/test/java/com/sebastian_daschner/jaxrs_analyzer/analysis/ProjectAnalyzerTest.java
@@ -210,6 +210,11 @@ public class ProjectAnalyzerTest {
         ResourceMethod fourthGet = ResourceMethodBuilder.withMethod(HttpMethod.GET, "Returns a test string with plain text.").andAcceptMediaTypes("application/json")
                 .andResponseMediaTypes("text/plain", "application/json").andResponse(200, ResponseBuilder.withResponseBody(stringIdentifier).build()).build();
         addMethods(resources, "test/test", fourthGet);
+        
+        // resourceWithoutClassLevelJavadoc/test
+        ResourceMethod getResourceWithoutJavadoc = ResourceMethodBuilder.withMethod(HttpMethod.GET, "Returns a test string in json.").andAcceptMediaTypes("application/json")
+                .andResponseMediaTypes("application/json").andResponse(200, ResponseBuilder.withResponseBody(stringIdentifier).build()).build();
+        addMethods(resources, "resourceWithoutJavadoc/test", getResourceWithoutJavadoc);
 
         // complex
         ResourceMethod eighthGet = ResourceMethodBuilder.withMethod(HttpMethod.GET).andResponseMediaTypes("application/json")

--- a/src/test/jaxrs-test/com/sebastian_daschner/jaxrs_test/ResourceWithoutClassLevelJavadoc.java
+++ b/src/test/jaxrs-test/com/sebastian_daschner/jaxrs_test/ResourceWithoutClassLevelJavadoc.java
@@ -1,0 +1,50 @@
+/*
+ * Copyright (C) 2015 Sebastian Daschner, sebastian-daschner.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.sebastian_daschner.jaxrs_test;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Stateless
+@Path("resourceWithoutJavadoc")
+public class ResourceWithoutClassLevelJavadoc {
+
+    /**
+     * Simple field comment (see issue #178)
+     */
+    @Inject
+    Manager<Integer> manager;
+
+    /**
+     * Returns a test string in json.
+     *
+     * @return Ignore this comment
+     */
+    @GET
+    @Path("test")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response testJson() {
+        return Response.ok("{\"hi\":\"hello\"}", MediaType.APPLICATION_JSON).build();
+    }
+
+}

